### PR TITLE
Performance fixes

### PR DESCRIPTION
**Pull request by JeffDotPng**
Small PR to improve performance again.
Changes:
- Fixed sidebar rerendering constantly again
- Localised `ControlBar` rerenders to specific buttons

### CODE REVIEW

**Review by itsjoeoui — APPROVED** [review]
lgtm

File: src/feature/flow_chart_panel/FlowChartTabView.tsx [hunk 1]
```diff
@@ -98,7 +97,7 @@ const FlowChartTab = () => {
   const addNewNode = useAddNewNode(setNodes, getNodeFuncCount);
   const sidebarCustomContent = useMemo(
     () => <SidebarCustomContent onAddNode={addNewNode} />,
-    [nodes, edges]
+    []
```
Comment (smahmed776) [inline src/feature/flow_chart_panel/FlowChartTabView.tsx:100]: `addNewNode` is dependent on `nodes` and `edges` variables as it takes `getNodeFuncCount` as a parameter that needed to be updated each time `nodes` and `edges` get updated. So it's required to add them to the dependency list. Or we have to find another alternative for this.

**Review by smahmed776 — CHANGES_REQUESTED** [review]

File: src/feature/flow_chart_panel/FlowChartTabView.tsx (hunk 1 above)
Comment (JeffDotPng) [reply src/feature/flow_chart_panel/FlowChartTabView.tsx]: We can just observe `addNewNode` for changes, works perfectly.

**Review by smahmed776 — COMMENTED** [review]
Looks goods!

**Review by smahmed776 — APPROVED** [review]

**Pull request merged**